pilot/pkg/bootstrap/server.go

```diff
@@ -76,6 +81,14 @@ var (
 		plugin.Health,
 		plugin.Mixer,
 	}
+
+	IstiodAddress = env.RegisterStringVar("ISTIOD_ADDR", "",
+		"Local service address of istiod. Default value is mesh.defaultConfig.discoveryAddress, use this to override if the discovery address is external")
```

Comment:
why do we need this?  This is for the proxy and not for Pilot.

Reply:
Istiod needs to know its own name - so it can get the proper certs from K8S. 

Unlike old istio, we can run Istiod with multiple service names ( and associated deployments ), for multi-tenant and for canary / controlled control plane rollouts. It works because we no longer 
depend on Citadel to give us DNS certs for injection. But we still need to know what is our own name. 

Now: we can actually get this from mesh.config - and that's the default we are using. However in some cases ( multi-cluster, or delployments with only injector and control plane managed extenally ) - istiod and discoveryAddress may be different. That's when this config will be used.

Reply:
And the plan is to add this to mesh config - together with caAddr, which will also default to 
discoveryAddress ( and name for DNS ), but can be external as well. 

However API changes require more thinking - and I don't want to rush to add APIs until 
we have a proof that the model works.

Reply:
Moved to features, added explanation.